src/hardware_manager.rs

```diff
@@ -1,24 +1,14 @@
 use lazy_static::lazy_static;
 use std::convert::From;
-use std::sync::Mutex;
-
-struct NavigationManager {
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+pub struct NavigationManager {
     navigator: navigator_rs::Navigator,
+    sentinel: Option<std::thread::JoinHandle<()>>,
 }
-
-lazy_static! {
-    static ref NAVIGATOR: Mutex<Option<NavigationManager>> = Mutex::new(None);
-}
-
-impl NavigationManager {
-    fn get_instance() -> &'static Mutex<Option<Self>> {
-        if NAVIGATOR.lock().unwrap().is_none() {
-            *NAVIGATOR.lock().unwrap() = Some(NavigationManager {
-                navigator: navigator_rs::Navigator::new(),
-            });
-        }
-        &NAVIGATOR
-    }
+#[derive(Debug, Clone, Default)]
+pub struct Data {
+    pub state: navigator_rs::SensorData,
```

Comment:
Do these need to be pub?

Reply:
I see it is being used as the return type of Data::read, but Data::read might not need to be exposed

Reply:
it needn't, the functions use it from same module and return derived values, ok